Model/Person/Tagging/Builder.php

```diff
@@ -37,57 +37,185 @@
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Nosto\Tagging\CustomerData\HashedTagging;
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Nosto\Tagging\Logger\Logger as NostoLogger;
 
 /**
  * Builder class for buyer
  * @package Nosto\Tagging\Model\Order\Buyer
  */
 class Builder extends PersonBuilder
 {
+
+    const GENDER_MALE = 'Male';
+    const GENDER_FEMALE = 'Female';
+    const GENDER_MALE_ID = '1';
+    const GENDER_FEMALE_ID = '2';
+
+    private $groupRepository;
+    private $customerRepository;
+    private $logger;
+
+    /**
+     * Builder constructor.
+     * @param GroupRepository $groupRepository
+     * @param CustomerRepositoryInterface $customerRepository
+     * @param NostoEmailRepository $emailRepository
+     * @param EventManager $eventManager
+     * @param NostoHelperData $nostoHelperData
+     */
+    public function __construct(
+        GroupRepository $groupRepository,
+        CustomerRepositoryInterface $customerRepository,
+        NostoEmailRepository $emailRepository,
+        NostoLogger $logger,
+        EventManager $eventManager,
+        NostoHelperData $nostoHelperData
+    ) {
+        $this->groupRepository = $groupRepository;
+        $this->customerRepository = $customerRepository;
+        $this->logger = $logger;
+        parent::__construct($emailRepository, $eventManager, $nostoHelperData);
+    }
+
     /**
      * @inheritdoc
+     * @return Customer
      */
     public function buildObject(
         $firstName,
         $lastName,
         $email,
         $phone = null,
         $postCode = null,
-        $country = null
+        $country = null,
+        $customerGroup = null,
+        $dateOfBirth = null,
+        $gender = null,
+        $customerReference = null
     ) {
-        $user = new User();
-        $user->setFirstName($firstName);
-        $user->setLastName($lastName);
-        $user->setEmail($email);
-        $user->setPhone($phone);
-        $user->setPostCode($postCode);
-        $user->setCountry($country);
-
-        return $user;
+        $customer = new Customer();
+        $customer->setFirstName($firstName);
+        $customer->setLastName($lastName);
+        $customer->setEmail($email);
+        $customer->setPhone($phone);
+        $customer->setPostCode($postCode);
+        $customer->setCountry($country);
+        $customer->setCustomerGroup($customerGroup);
+        $customer->setCustomerReference($customerReference);
+        $customer->setGender($gender);
+        if ($dateOfBirth !== null) {
+            $customer->setDateOfBirth(\DateTime::createFromFormat("Y-m-d", $dateOfBirth));
+        }
+
+        return $customer;
     }
 
     /**
      * Builds person from the current session / logged in user
      *
      * @param CurrentCustomer $currentCustomer
-     * @return AbstractPerson|null
+     * @return Customer|null
```

Comment:
The annotation on the builder function returns an object of type `AbstractPerson`. Please double check with the debugger and fix the annotations.